File: ps1/cash.c
// Returns how many coins are needed when giving change

// Imports
#include <cs50.h>
#include <stdio.h>
#include <math.h>

// Main code
int main(void)
{
    
    // Save user input
    float dollars = 0.00;
    
    // Prompt user for change input
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0.009);
    
    // Convert float to integer (dollars to cents)
    // Save the number of coins needed
    int cents = round(dollars * 100), num_coins = 0;
    
    // Pay with largest possible coins, increment coin count
    while (cents >= 25)
    {
        cents -= 25;
        num_coins ++;
    }
    
    while (cents >= 10)
    {
        cents -= 10;
        num_coins ++;
    }
    
    while (cents >= 5)
    {
        cents -= 5;
        num_coins ++;
    }
    
    while (cents >= 1)
    {
        cents -= 1;
        num_coins ++;
    }
    // Print number of coins needed
    printf("%i\n", num_coins); 
}

File: pset6/cash/cash.py
# calculates coins for given amount of change
from cs50 import get_float

# get user input
while True:
    dollars = get_float("Change owed: ")
    if dollars >= 0:
        break
# convert dollars to cents
cents = int(dollars*100)
# keep track of coins/change needed
num_coins = 0
# give greedy coins until we have paid all change
while cents > 0:
    if cents // 25 > 0:
        num_coins += 1
        cents -= 25
    elif cents // 10 > 0:
        num_coins += 1
        cents -= 10
    elif cents // 5 > 0:
        num_coins += 1
        cents -= 5
    elif cents // 1 > 0:
        num_coins += 1
        cents -= 1
# return the total number of coins needed
print(num_coins, "\n")